src/engine/controls/cuecontrol.cpp

```diff
@@ -579,14 +582,19 @@ void CueControl::hotcueSet(HotcueControl* pControl, double v) {
     pCue->setHotCue(hotcue);
     pCue->setLabel("");
     pCue->setType(Cue::Type::HotCue);
-    // TODO(XXX) deal with spurious signals
-    attachCue(pCue, pControl);
 
-    if (getConfig()->getValue(ConfigKey("[Controls]", "auto_hotcue_colors"), false)) {
-        const QList<PredefinedColorPointer> predefinedColors = Color::kPredefinedColorsSet.allColors;
-        pCue->setColor(predefinedColors.at((hotcue % (predefinedColors.count() - 1)) + 1));
+    auto hotcueColorPalette = m_colorPaletteSettings.getHotcueColorPalette();
+    ConfigKey autoHotcueColorsKey("[Controls]", "auto_hotcue_colors");
+    if (getConfig()->getValue(autoHotcueColorsKey, false)) {
+        auto colors = hotcueColorPalette.m_colorList;
+        pCue->setColor(colors.at(hotcue % colors.count()));
+    } else {
+        pCue->setColor(hotcueColorPalette.m_colorList.first());
```

Comment:
Yes I agree, I’ll handle this in the mext PR. I’ll create the concept of “default cue color strategy”. I will add two strategies:
- Assign different color to each cue (as currently implemented)
- Use default color (as done in the code you commented, but with the default color configurable in preferences)

We’ll need to think on what’s the best way to do “different color per cue type”

Reply:
Again, no. Black is an ugly and awkward color for a hotcue. There is no need for a concept of "no color".

Reply:
In my next PR this line will disappear. Users will be able to choose the default color so there’s no need to discuss this.

Reply:
Can we choose black here?
Than it behaves exactly like 2.2 in the mean time. For the future PR we don't have to deal with red cues where the user also not has set a color.

@Be-ing : Black is only a placeholder for "no color". We can't discuss this away, because all cues have no color in 2.2.

Reply:
Black is simply not okay here. In 3 out of 4 skins black hotcues are hard to see on the waveform.

> In my next PR this line will disappear. Users will be able to choose the default color so there’s no need to discuss this.

Customizing this will be good. But we still need a default for that option, so this issue will still remain.

Reply:
We need an alternative color that is not in the palette to distinguish legacy cues from cues with an explicit color set. Black is Ok since this is only a place holder and need to be replaced for displaying. Maybe transparent is a better choice?

Reply:
Again, no, there is no need for distinguishing cues set in earlier versions of Mixxx. Again, we are working on this now, holding back the 2.3 release even further, because the old implementation was overcomplicated. Let's not keep adding unnecessary complications.